lib/iris/_lazy_data.py

```diff
@@ -596,8 +599,14 @@ def map_complete_blocks(src, func, dims, out_sizes, *args, **kwargs):
         for dim, size in zip(dims, out_sizes):
             out_chunks[dim] = size
 
-        result = data.map_blocks(
-            func, *args, chunks=out_chunks, dtype=src.dtype, **kwargs
-        )
+        # Assume operation does not change dtype and meta if not specified.
+        if "meta" not in kwargs:
+            kwargs["meta"] = da.utils.meta_from_array(data)
+        if "dtype" in kwargs:
+            kwargs["meta"] = kwargs["meta"].astype(kwargs["dtype"])
+        else:
+            kwargs["dtype"] = kwargs["meta"].dtype
+
+        result = data.map_blocks(func, *args, chunks=out_chunks, **kwargs)
```

Comment:
I'm finding this special handling of "dtype" and/or "meta" rather confusin,.
And I note that it  is currently **_only_** exercised in tests, and nowhere in the actual code.

So I'm wondering if we can do without ??

If not, maybe they need documenting as a special case, or even listing as explicit keys :  in the current code, it is rather confusing that these keywords, which specifically control `map_blocks`, are  are part of kwargs but  **_not_** passed down to `func` like the docstring says they are (!).

Reply:
I was concerned about the assumption that the output `dtype` is always the same as the input `dtype`, that's why I thought it might be good to be able to specify this. I checked the `dtypes` coming out of the functions that use `map_complete_blocks` and tried to specify those in c1e5e1a. Does it look better now?